performance/autoupdate-clients/main.go

```diff
@@ -0,0 +1,258 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"log"
+	"net/http"
+	"net/http/cookiejar"
+	"net/url"
+	"os"
+	"os/signal"
+	"sort"
+	"strconv"
+	"sync"
+	"time"
+)
+
+func main() {
+	N := getNFromCommandLine()
+	host, user, pw := getParams()
+	jar := getCookieJar(user, pw, host)
+
+	sumBytes := 0
+	oldSumKB := 0
+	changeIds := make(map[int]int)
+	changeIdsMutex := &sync.Mutex{}
+	gotUpdate := false
+
+	interrupt := make(chan os.Signal, 1)
+	signal.Notify(interrupt, os.Interrupt)
+
+	quit := make(chan int)
+	recv := make(chan Message)
+	for i := 0; i < N; i++ {
+		go client(quit, recv, i, jar, host)
+		time.Sleep(100 * time.Millisecond)
+	}
+
+	go func(quit chan int) {
+		for {
+			select {
+			case <-quit:
+				return
+			case <-time.After(time.Second):
+				if !gotUpdate {
```

Comment:
This is a race condition. The variable is written in line 98 in a separat goroutine.

You should test/run you program with the race detector: https://golang.org/doc/articles/race_detector.html

Use either a lock or a channel to communicate between the goroutines